### PR TITLE
Move schools-related attributes to Eligibility

### PR DESCRIPTION
**Pull request by tekin**
This continues the work to separate the common claim-related data
(identity, payment, payroll, etc) from the policy-specific fields,
making it easier to support more than one type of policy (i.e. Maths &
Physics).

This commit focuses on the two schools associations and the closely
associated employment_status field. Because all three attributes are
closely related and interdependent they've had to be migrated at the
same time.

Note that no effort has been made to migrate the data from existing
claims as we're not yet live so any existing data can simply be deleted.
The best way to do this on the Azure environments is probably going to
be to drop the database and perform a release to force the rebuilding of
the database.

## Temporary fixes and future work

- As with the preceding work that moved qts_award_year, a bunch of
  attributes are temporarily being delegated from the claim model to the
  eligibility model to keep these changes as small and focused as
  possible. These will need removing before Maths & Physics can be
  introduced as at that point, the claim model will need to be entirely
  generic and not know anything about the policy-specific eligibility
  fields.
- The ClaimUpdate model has needed to be updated with knowledge specific
  both to the claim model and the eligibility model. The code doesn't
  look great at the moment, but we know it will need adjusting at the
  point the new policy is introduced, and that feels like a better point
  to figure out the best way to model the behaviour for adjusting
  in-progress claims (and their eligibility) and their dependent
  attributes
- Same goes for many of the specs; many have had to be updated with
  student loans-specific attributes and knowledge, in some cases
  coupling them to the specifics of student loans. As above, we'll be in
  a better place to try and untangle these at the point the new policy
  is introduced.

Part of https://trello.com/c/9gpS69mI/536-1-separate-an-eligibility-model-from-the-claim-model

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -103,15 +106,12 @@
     t.integer "student_loan_plan"
     t.string "eligibility_type"
     t.uuid "eligibility_id"
-    t.index ["claim_school_id"], name: "index_tslr_claims_on_claim_school_id"
-    t.index ["current_school_id"], name: "index_tslr_claims_on_current_school_id"
     t.index ["eligibility_type", "eligibility_id"], name: "index_tslr_claims_on_eligibility_type_and_eligibility_id"
-    t.index ["employment_status"], name: "index_tslr_claims_on_employment_status"
```
Comment (erbridge) [inline db/schema.rb]: Did we intentionally lose this index? I think it makes sense, but just wanted to check.

Comment (tekin) [reply db/schema.rb]: Interesting, had no idea the original column had an index. I don't think we actually need it on the new table, it certainly doesn't feel like an attribute we're likely to query against any time soon.

Comment (tekin) [reply db/schema.rb]: lol, looks like myself and @mec added the original index..

File: app/controllers/admin/claims_controller.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 class Admin::ClaimsController < Admin::BaseAdminController
   def index
-    claims = TslrClaim.includes(:claim_school, :current_school, :eligibility).submitted.order(:submitted_at)
+    claims = TslrClaim.includes(eligibility: [:claim_school, :current_school]).submitted.order(:submitted_at)
```
[the remark this answers is not in the document]
Comment (tekin) [reply app/controllers/admin/claims_controller.rb]: Do you mean Bullet?

[the remark this answers is not in the document]
Comment (tekin) [reply app/controllers/admin/claims_controller.rb]: ah right, yep, we'll need to update it.

[the remark this answers is not in the document]
Comment (erbridge) [reply app/controllers/admin/claims_controller.rb]: No, I mean `config/brakeman.ignore` which references this line.

[the remark this answers is not in the document]
Comment (tekin) [reply app/controllers/admin/claims_controller.rb]: done!

**Comment by tekin** [comment]
@erbridge is this good now?

File: spec/models/student_loans/eligibility_spec.rb [hunk 3]
```diff
@@ -48,13 +95,50 @@
...
+  end
+
   context "when saving in the “submit” context" do
+    it "is valid when all attributes are present" do
+      expect(build(:student_loans_eligibility, :submittable)).to be_valid(:submit)
+    end
+
     it "is not valid without a value for qts_award_year" do
-      expect(StudentLoans::Eligibility.new).not_to be_valid(:submit)
+      expect(build(:student_loans_eligibility, :submittable, qts_award_year: nil)).not_to be_valid(:submit)
```
Comment (robbiepaul) [inline spec/models/student_loans/eligibility_spec.rb]: I prefer this 👍

**Comment by tekin** [comment]
Holding off on merging this until after https://github.com/DFE-Digital/dfe-teachers-payment-service/pull/221 has been merged to deal with merge conflicts here, rather than there.

**Pull request merged**